src/test/java/org/cactoos/time/DateAsTextTest.java

```diff
@@ -0,0 +1,168 @@
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.cactoos.time;
+
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.util.Calendar;
+import java.util.Locale;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.Matchers;
+import org.junit.Test;
+
+/**
+ * Tests for DateAsText.
+ * @author Sven Diedrichsen (sven.diedrichsen@gmail.com)
+ * @version $Id$
+ * @since 1.0
+ * @checkstyle JavadocMethodCheck (500 lines)
+ * @checkstyle MagicNumberCheck (500 lines)
+ */
+@SuppressWarnings("PMD.AvoidDuplicateLiterals")
+public class DateAsTextTest {
+
+    @Test
+    public final void testZonedDateTimeFormattedAsIsoDateTime()
+        throws Exception {
+        final ZonedDateTime date = ZonedDateTime.of(
+            2017, 12, 13, 14, 15, 16, 17, ZoneId.of("Europe/Berlin")
+        );
+        MatcherAssert.assertThat(
+            "Can't format a ZonedDateTime with default/ISO format.",
+            new DateAsText(date).asString(),
+            Matchers.is("2017-12-13T14:15:16.000000017+01:00[Europe/Berlin]")
+        );
+    }
+
+    @Test
+    public final void testZonedDateTimeFormattedWithFormatString()
+        throws Exception {
+        final ZonedDateTime date = ZonedDateTime.of(
+            2017, 12, 13, 14, 15, 16, 17, ZoneId.of("Europe/Berlin")
+        );
+        MatcherAssert.assertThat(
+            "Can't format a ZonedDateTime with format.",
+            new DateAsText(date, "yyyy-MM-dd HH:mm:ss").asString(),
+            Matchers.is("2017-12-13 14:15:16")
+        );
+    }
+
+    @Test
+    public final void testZonedDateTimeFormattedWithFormatStringWithLocale()
+        throws Exception {
+        final ZonedDateTime date = ZonedDateTime.of(
+            2017, 12, 13, 14, 15, 16, 17, ZoneId.of("Europe/Berlin")
+        );
+        MatcherAssert.assertThat(
+            "Can't format a ZonedDateTime with format using locale.",
+            new DateAsText(date, "yyyy MMM dd. HH.mm.ss", Locale.FRENCH)
+                .asString(),
+            Matchers.is("2017 déc. 13. 14.15.16")
+        );
+    }
+
+    @Test
+    public final void testDateFormattedUsingIsoFormatter() throws Exception {
+        final Calendar calendar = Calendar.getInstance();
+        calendar.set(2017, Calendar.DECEMBER, 13, 14, 15, 16);
+        calendar.set(Calendar.MILLISECOND, 17);
+        MatcherAssert.assertThat(
+            "Can't format a java.util.Date with ISO format.",
+            new DateAsText(calendar.getTime()).asString(),
+            Matchers.is("2017-12-13T14:15:16.017")
```

Comment:
@svendiedrichsen why there is no time zone in this text?

Reply:
There is no timezone because java.util.Date does not carry one. It does implicitly reference UTC timezone though. So I have used ZonedDateTime with UTC timezone to create a string from it. This will lead to shifting time in the string if the Date provided was created with a different timezone. Funny enough does it know with which timezone it has been created.